### DISCUSSION TITLE
Run Eslint with Bazel

### DISCUSSION THREAD

**Discussion by emresiimseek** [bazelbuild/rules_nodejs#3633] [Q&A]
Hey everyone, 

Is there any way to run "lint": "eslint --ext .js,.ts,.jsx,.tsx src" script with Bazel?
I could not find anything on google.

Thanks.

**Comment by alexeagle** [comment]
rules_nodejs is no longer maintained.

I started some work on "proper" eslint support under Bazel: https://github.com/aspect-build/rules_js/pull/658
But you can also just run the eslint binary like any other tool: https://docs.aspect.build/rules/aspect_rules_js/docs/#using-binaries-published-to-npm

**Comment by emresiimseek** [comment]
Thanks for your response Alex.
I installed eslint package with yarn and I loaded eslint package in BUILD file but I could not run my eslint lint. What am I missing?

```
load("@npm//:eslint/package_json.bzl", eslint_bin = "bin")
eslint_bin.eslint(
    name = "lint",
    srcs = ["src/types/message.ts"],
    args = [
        ".eslintrc.yaml",
    ],
)
```

**Reply by alexeagle** [reply]
Sorry, that's not enough information, it looks fine on a visual inspection.

**Comment by ysshah** [comment]
Something like this is what ended up working for me:

```bazel
load("@npm//:eslint/package_json.bzl", eslint_bin = "bin")

eslint_bin.eslint_test(
    name = "lint",
    chdir = package_name(),
    data = glob(["src/**/*.ts", "src/**/*.tsx"]) + [".eslintrc.cjs"],
    args = ["."],
)
```

And you can swap out `eslint_test` with `eslint_binary` for a run target instead of a test target.

**Comment by alexeagle** [comment]
I made a new ruleset rules_lint that provides first-class support for this: https://github.com/aspect-build/rules_lint/blob/main/docs/eslint.md
